core/src/main/scala/tech/beshu/ror/configuration/SslConfiguration.scala

```diff
@@ -83,7 +83,8 @@ final case class SslConfiguration(keystoreFile: JFile,
                                   keyAlias: Option[SslConfiguration.KeyAlias],
                                   allowedProtocols: Set[SslConfiguration.Protocol],
                                   allowedCiphers: Set[SslConfiguration.Cipher],
-                                  verifyClientAuth: Boolean)
+                                  clientAuthenticationEnabled: Boolean,
+                                  certificateVerificationEnabled: Boolean)
```

Comment:
this class is used to describe external SSL and internode SSL. At the moment certificate verification can be disabled only for internode SSL. Maybe we should allow user to disable it also for external SSL (eg. self signed cert usage). 

@sscarduzio @pondzix WDYT?

Reply:
Definitely we should set the HTTPS not to verify client cert by default. Otherwise admins should distribute certificates to all the clients before they can connect. Client cert verification is not expected by default in HTTPS, and this is also the current behaviour. So it should continue to be like this, or would break all customers' upgrades.

Reply:
I'm not talking about defaults but possibility to use self signed certs. Seems that at the moment you cannot do that. Default should be the same as now

Reply:
I split `SslConfiguration` to `InternodeSslConfiguration` and `ExternalSslConfiguration` to make it more explicit